src/main/java/seedu/address/model/visit/Visit.java

```diff
@@ -66,6 +68,12 @@ public boolean isSameLocationId(Index personId) {
         return personId == this.personId;
     }
 
+    /**
+     * Returns Predicate for filtering out the infected visits from all visits
+     */
+    public static Predicate<Visit> getPredicateForInfectedVisits(HashSet<Index> infectedIds) {
+        return visit -> infectedIds.contains(visit.getPersonId());
+    }
```

Comment:
Consider putting this under ModelPredicate as well?

Reply:
Done.